addons/website_sale/views/templates.xml

```diff
@@ -1756,26 +1742,26 @@
                         </div>
                     </div>
                     <div class="row">
-                        <div class="col-12 col-xl-auto order-xl-2 d-none d-xl-block">
+                        <div class="col-12 offset-xl-1 col-lg-5 col-xl-4 order-lg-2">
                             <t t-call="website_sale.cart_summary">
                                 <t t-set="hide_promotions">True</t>
                                 <t t-set="redirect" t-valuef="/shop/address"/>
                             </t>
                         </div>
-                        <div class="col-12 col-xl order-xl-1 oe_cart">
+                        <div class="col-12 col-lg-7 order-lg-1 oe_cart">
                             <div>
                                 <t t-if="mode == ('new', 'billing')">
-                                    <h2 class="o_page_header mt8">
+                                    <h3 class="mb-3">
```

Comment:
General note:  avoid customizing inline elements margin unless it's strictly necessary.
Let's keep in mind that the the user can change the font-family... if I switch to _Lobster_, for example, I would probably adapt margins accordingly... but these classes will prevent any change ;)

I'll let you check which margin adaptations are necessary